server/src/main/java/org/opensearch/index/remote/RemoteStoreUtils.java

```diff
@@ -116,4 +122,42 @@ static String longToUrlBase64(long value) {
         String base64Str = Base64.getUrlEncoder().encodeToString(hashBytes);
         return base64Str.substring(0, base64Str.length() - 1);
     }
+
+    /**
+     * Returns <code>true</code> iff current cluster settings have:
+     * <br>
+     * - <code>remote_store.compatibility_mode</code> set to <code>mixed</code>
+     * <br>
+     * - <code>migration.direction</code> set to <code>remote_store</code>
+     * <br>
+     * <code>false</code> otherwise
+     *
+     * @param settings Current Cluster Settings
+     * @return <code>true</code> or <code>false</code> depending upon the above-mentioned condition
+     */
+    public static boolean ongoingDocrepToRemoteMigration(Settings settings) {
+        return RemoteStoreNodeService.REMOTE_STORE_COMPATIBILITY_MODE_SETTING.get(
+            settings
+        ) == RemoteStoreNodeService.CompatibilityMode.MIXED
+            && RemoteStoreNodeService.MIGRATION_DIRECTION_SETTING.get(settings) == RemoteStoreNodeService.Direction.REMOTE_STORE;
+    }
+
+    /**
+     * Fetches segment and translog repository names from remote store node attributes
+     * @param discoveryNodes Current set of {@link DiscoveryNodes} in the cluster
+     * @return {@link Tuple} with segment repository name as first element and translog repository name as second element
+     */
+    public static Tuple<String, String> getRemoteStoreRepositoryNames(DiscoveryNodes discoveryNodes) {
```

Comment:
The function doesn't fit well . Tuple also doesn't look the right datastructure for this . Would prefer two functions for retrieving translog repo and segment repo.